scripts/extend_index_yaml_test.py

```diff
@@ -0,0 +1,372 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for scripts/extend_index_yaml.py."""
+
+from __future__ import annotations
+
+import tempfile
+
+from core.tests import test_utils
+
+from . import extend_index_yaml
+
+
+class ExtendIndexYamlTest(test_utils.GenericTestBase):
+    """Class for testing the extend_index_yaml script."""
+
+    def setUp(self) -> None:
+        super(ExtendIndexYamlTest, self).setUp()
+        self.index_yaml_file = tempfile.NamedTemporaryFile()
+        self.web_inf_index_yaml_file = tempfile.NamedTemporaryFile()
+        self.index_yaml_file_name = self.index_yaml_file.name
+        self.web_inf_index_yaml_file_name = self.web_inf_index_yaml_file.name
+        self.index_yaml_swap = self.swap(
+            extend_index_yaml, 'INDEX_YAML_PATH',
+            self.index_yaml_file.name)
+        self.web_inf_index_yaml_swap = self.swap(
+            extend_index_yaml, 'WEB_INF_INDEX_YAML_PATH',
+            self.web_inf_index_yaml_file.name)
+        self.open_index_yaml_r = open(
+            self.index_yaml_file.name, 'r', encoding='utf-8')
+        self.open_index_yaml_w = open(
+            self.index_yaml_file.name, 'w', encoding='utf-8')
+        self.open_web_inf_index_yaml = open(
+            self.web_inf_index_yaml_file.name, 'a', encoding='utf-8')
+
+    def tearDown(self) -> None:
+        super(ExtendIndexYamlTest, self).tearDown()
+        self.index_yaml_file.close()
+        self.web_inf_index_yaml_file.close()
+
+    def test_extend_index_yaml_with_changes(self):
+        index_yaml = """indexes:
+
+- kind: AppFeedbackReportModel
+  properties:
+  - name: created_on
+  - name: scrubbed_by
+
+- kind: BlogPostRightsModel
+  properties:
+  - name: blog_post_is_published
+  - name: editor_ids
+  - name: last_updated
+    direction: desc
+"""
+        web_inf_index_yaml = """indexes:
+
+- kind: ClassifierTrainingJobModel
+  properties:
+  - name: status
+  - name: next_scheduled_check_time
+"""
+        expected_index_yaml = """indexes:
+
+- kind: AppFeedbackReportModel
+  properties:
+  - name: created_on
+  - name: scrubbed_by
+
+- kind: BlogPostRightsModel
+  properties:
+  - name: blog_post_is_published
+  - name: editor_ids
+  - name: last_updated
+    direction: desc
+
+- kind: ClassifierTrainingJobModel
+  properties:
+  - name: status
+  - name: next_scheduled_check_time
+"""
+
+        with self.index_yaml_swap, self.web_inf_index_yaml_swap:
+            with self.open_index_yaml_w as f:
+                f.write(index_yaml)
+            with self.open_web_inf_index_yaml as f:
+                f.write(web_inf_index_yaml)
```

Comment:
Can you put this in a helper function? `_fill_indexes(index_yaml, web_inf_index_yaml)`

Reply:
Or looking at it now all the test files are basically the same, so you can put the whole test code into a helper function and the only call it with different indexes.

Reply:
Will got with the latter.
Done